core/src/main/scala/kafka/server/DelegationTokenManager.scala

```diff
@@ -257,6 +153,13 @@ class DelegationTokenManager(val config: KafkaConfig,
     scramCredentialMap.toMap
   }
 
+  /**
+   * @param token
+   */
+  def updateToken(token: DelegationToken): Unit = {
```

Comment:
> Should we protect this as well with the lock? The ZK variant is protected through the create/renew/expire methods but this isn't. (As I see we only use it through `DelegationTokenPublisher.onMetadataUpdate` but I don't know if calls to that come from multiple threads or just a single one.)

The calls come from just a single thread. They have to because replay of records out of order could lead to inconsistent state.